consensus/istanbul/core/core.go

```diff
@@ -377,7 +378,13 @@ func (c *core) commit() error {
 			c.waitForDesiredRound(nextRound)
 			return nil
 		}
-		if err := c.backend.Commit(proposal, aggregatedSeal, aggregatedEpochValidatorSetSeal); err != nil {
+
+		result, isCached := c.current.GetStateProcessResult(proposal.Hash())
+		if !isCached {
```

Comment:
in case we don't find in the cache (for example, if nodes goes offline and then online again); we should call `Verify()` again, instead of failing